crates/wasmtime/src/store.rs

```diff
@@ -1129,7 +1188,9 @@ impl StoreOpaque {
             // If this finished successfully then we were resumed normally via a
             // `poll`, so inject some more fuel and keep going.
             Ok(()) => {
-                self.add_fuel(fuel_to_inject).unwrap();
+                if fuel_to_inject > 0 {
```

Comment:
Can this be passed as an `Option` instead of 0-or-nonzero?

Also can the documentation for this function be updated to indicate that might be used for non-fuel-based-scenarios? (perhaps renaming the function as well)

Reply:
Fixed -- this is now `async_yield_impl` and the fuel update happens in `out_of_gas` (one of its callers) instead.